third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/BarrierOpToLLVM.cpp

```diff
@@ -120,18 +121,35 @@ struct BarrierExpectConversion
                   ConversionPatternRewriter &rewriter) const override {
     Location loc = op->getLoc();
     auto b = TritonLLVMOpBuilder(loc, rewriter);
+    auto barrierTy = op.getAlloc().getType();
     auto smemObj = LLVM::getSharedMemoryObjectFromStruct(
         loc, adaptor.getAlloc(),
-        typeConverter->convertType(op.getAlloc().getType().getElementType()),
-        rewriter);
+        typeConverter->convertType(barrierTy.getElementType()), rewriter);
+    // If several CTAs cast to the same barrier, that barrier will receive all
+    // the bytes from its broadcast group
+    auto numCTAs = triton::gpu::lookupNumCTAs(rewriter);
+    auto expectedBytes = op.getSize() * (numCTAs / barrierTy.getNumElements());
```

Comment:
What is `numCTAs` doing here? Don't you need to divide by `num_cta_split` as `nbytes_per_cta` does?

Reply:
This function already gets the nbytes_per_cta. Now we need to multiply that by the number of CTAs that broadcast.

Reply:
Note that this is the number of CTAs that broadcast **in the barrier**, as opposed to the descriptor. This tells you how many CTAs will send their bytes to the leader CTA.